crates/mirror-mirror/src/type_info/mod.rs

```diff
@@ -1201,6 +1201,7 @@ impl<'a> OpaqueType<'a> {
     }
 }
 
+/// A superset of `Type` that can also describe `Variant`s.
```

Comment:
Just some thoughts about `TypeAtPath` in general, not specific to this change:

Probably worth revisiting `TypeAtPath` at some point. I remember it being a little clunky to use when I was using it our game.

How there are types for variants but not values is an annoying lack of symmetry. We have an `Enum` trait, but not a `Variant` trait. Not sure how to clean it up though.